### ISSUE TITLE
Bug Local time setting is not shown

### ISSUE THREAD

**Issue by Gilletarom** [rgleason/weather_routing_pi#144]
Unless I'm mistaken, WR always calculates times in UTC internally. BUT schedules can be displayed in two ways:
-utc
- local time

The choice is made in the settings:

![2023-09-10 11_01_39-Weather-routing - WeatherRoutingConfiguration xml](https://github.com/rgleason/weather_routing_pi/assets/6445543/85b73e2f-b762-4e0a-a339-3df2705f337d)

Normally, this choice applies to all times, for all search configurations.

BUT, in version 1.15.11, this choice is not respected for the display for new searches. The correction must be carried out manually, in contradiction with the parameter settings.

![2023-09-10 10_53_33-](https://github.com/rgleason/weather_routing_pi/assets/6445543/69184854-522a-454c-8eaf-47033ee9cb7b)
![2023-09-10 10_53_12-](https://github.com/rgleason/weather_routing_pi/assets/6445543/1afa4fc9-3501-41ec-9bfe-bc45652c532f)

Can this be corrected by ensuring that the choice made in the settings is respected?

**Comment by sebastien-rosset** [comment]
> Normally, this choice applies to all times, for all search configurations.
> BUT, in version 1.15.11, this choice is not respected for the display for new searches. The correction must be carried out manually, in contradiction with the parameter settings.

No, this setting only applies to UI widgets that are controlled by the WR plugin. 

Currently:
- There is no centralized setting for date/time format. That RFE is tracked in https://github.com/OpenCPN/OpenCPN/pull/4052.
- OpenCPN core and each plugin may have their own settings for displaying time. Some plugins have no such controls, some plugins offer UTC and local time, and at least one plugin offers UTC, local time or LMT.

AFAIK, the screenshots you have pasted are the route panel, which belongs to the OpenCPN core code base. The date/times in that table are not controlled by the WR plugin.

Here is the sequence of tasks:
1. Get consensus on centralized settings of date/time formatting.
2. Merge PR (e.g. https://github.com/OpenCPN/OpenCPN/pull/4052) in OpenCPN.
3. Update the WR plugin to remove the date/time control, as it will be done centrally in the OpenCPN options, in a single location.

**Comment by Gilletarom** [comment]
Hello Sébastien Rosset,

Obviously, we encounter the same problem here as in other RFEs I've raised.

Thank you for detailing your response above.

Gilletarom

**Comment by sebastien-rosset** [comment]
> Hello Sébastien Rosset,
> 
> Obviously, we encounter the same problem here as in other RFEs I've raised.
> 
> Thank you for detailing your response above.
> 
> Gilletarom

PR https://github.com/OpenCPN/OpenCPN/pull/4052 has been merged, so now there is a step towards being able to specify the date time format in a single place (options -> display). Eventually the WR plugin will honor the global setting.

**Comment by rgleason** [comment]
Yes, unfortunately, unless you can internally check for the API and have separate code for date. 

The full global date feature won't be released until fall I think, but ask Dave or Leamas or Pavel. Its kind of when it is ready date.

**Comment by rgleason** [comment]
@sebastien-rosset  **Advises regardin Global Time (Local and GMT)**

> PR https://github.com/OpenCPN/OpenCPN/pull/4052 has been merged, so now there is a step towards being able to specify the date time format in a single place (options -> display). Eventually the WR plugin will honor the global setting.

However for now it cannot be accomplished until the Next Version of OpenCPN is released and certain changes are made in WR to utilize "global time".  The next OpenCPN release will not be for quite some time, perhaps Fall, so this is simply not something that can be done right now.

This is really an RFE since WR was never programmed for this.   You want global settings in OpenCPN to be used by the plugin. That is not possible now.

So lets close this. I am sure it will be done when the API being used in the next OpenCPN release has the necessary changes. 

@Gilletarom  I am going to close this for now to simplify this mess of Issues that we have to work through.  You may please reopen it when we get the new OpenPN Release.  ok?

**Comment by sebastien-rosset** [comment]
It would be a nice RFE in OpenCPN if the plugins could specify "this is the minimum version I require, and give me a more recent version if you can". That way we could have a single branch that enables features based on what is available.

**Comment by rgleason** [comment]
@sebastien-rosset 

The Plugin Manager allows only one version of a plugin in the master catalog, otherwise the "Green up arrow" on the right showing that there is an upgrade available, will not work or show.

Therefore unlike earlier practice, before we had the green up arrow signaling an upgrade available, we had to remove all mulitple plugin versions.

**Comment by Gilletarom** [comment]
Hello Sébastien Rosset,

I see this : "@Gilletarom I am going to close this for now to simplify this mess of Issues that we have to work through. You may please reopen it when we get the new OpenPN Release. ok?"

I understand that we have to wait for the next version of OpenCPN and a next version of WR. Obviously, it's a shame that we have to wait so long, but OpenCPN's time is running slower and slower, so we'll wait.

Gilletarom

**Comment by sebastien-rosset** [comment]
It's true there was a period without significant development in the weather routing plugin, though recently there has been an acceleration of improvements and bug fixes.
